src/tlp_lib/smartcontracts/EthereumSC.py

```diff
@@ -184,21 +199,61 @@ def _deploy_contract(
         """
         ContractFactory = self.web3.eth.contract(abi=abi, bytecode=bytecode)
 
-        contract = ContractFactory.constructor(
-            coins, start_time, list(map(int, extra_times)), list(map(int, upper_bounds)), helper_id
-        )
+        contract = ContractFactory.constructor()
 
-        tx_hash: HexBytes = contract.transact(
-            {"from": self.account, "value": functools.reduce(lambda x, y: x + y, coins)}
-        )
+        tx_hash: HexBytes = contract.transact({"from": self.account})
 
         tx_receipt: TxReceipt = self.web3.eth.wait_for_transaction_receipt(tx_hash)
 
         self._contract = self.web3.eth.contract(
             address=tx_receipt["contractAddress"], abi=abi
         )  # pyright: ignore[reportAttributeAccessIssue]
+
+        self._initialize_in_batches(coins, start_time, extra_times, upper_bounds, helper_id)
+
         return tx_receipt["contractAddress"]
 
+    def _initialize_in_batches(
+        self,
+        coins: SC_Coins,
+        start_time: int,
+        extra_times: SC_ExtraTime,
+        upper_bounds: SC_UpperBounds,
+        helper_id: int | ChecksumAddress,
+    ) -> None:
+        # Ensure that all lists have the same length
+        assert len(coins) == len(extra_times) == len(upper_bounds), "All input lists must have the same length"
+
+        total_parts = len(coins)
+        start_index = 0
+
+        while start_index < total_parts:
+            end_index = min(start_index + self._SC_PUZZLE_BATCH_SIZE, total_parts)
```

Comment:
ditto on slicing, but you can zip the batched lists, so you do

```py 
for coins_batch, extra_times_batch, upper_bounds_batch in zip(...)
```

Reply:
also changed this